### ISSUE TITLE
fix OAuth2Client

### ISSUE THREAD

**Issue by bclozel** [resthub/resthub-spring-stack#140]
@ncarlier found some bugs in our OAuth2 client implementation.
- The OAuth2Config bean [does not hold OAuth2 credentials](https://github.com/resthub/resthub-spring-stack/blob/master/resthub-web/resthub-web-client/src/main/java/org/resthub/web/Client.java#L183)
- The [retrieveAccessToken method ](https://github.com/resthub/resthub-spring-stack/blob/master/resthub-web/resthub-web-client/src/main/java/org/resthub/web/oauth2/OAuth2RequestFilter.java#L137) does not get the right credentials (refactoring problem probably)
- spring-security-oauth  has been released on central, so we can add it back to the stack
- and uncomment oauth tests

**Comment by sdeleuze** [comment]
OMG, they finally released it !!!

**Comment by bclozel** [comment]
@sdeleuze 
Sigh.
It seems the token is not getting properly serialized by spring-security-oauth2, although the [responsible class](https://github.com/SpringSource/spring-security-oauth/blob/master/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/common/OAuth2AccessTokenSerializer.java) looks good to me.

Got the following token in debug, which looks like the [actual OAuth2Token class](https://github.com/SpringSource/spring-security-oauth/blob/master/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/common/OAuth2AccessToken.java):

``` javascript
{"value":"adf83c57-830d-4861-b057-d9a6722b4f60",
"expiration":1353448198023,
"tokenType":"bearer",
"refreshToken":null,
"scope":[],
"additionalInformation":{},
"expired":false,
"expiresIn":1}
```

Why is this token not getting properly serialized?
Is there a conflict somewhere with [their jackson dependency](https://github.com/SpringSource/spring-security-oauth/blob/master/spring-security-oauth2/pom.xml#L143)?

**Comment by bclozel** [comment]
Here's a more detailed comment about this issue.

`spring-security-oauth 1.0.0.RELEASE` depends on:

``` xml
  <groupId>org.codehaus.jackson</groupId>
  <artifactId>jackson-mapper-asl</artifactId>
```

So that the [OAuth2AccessToken class](https://github.com/SpringSource/spring-security-oauth/blob/master/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/common/OAuth2AccessToken.java) can have a custom serializer [OAuth2AccessTokenSerializer](https://github.com/SpringSource/spring-security-oauth/blob/master/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/common/OAuth2AccessTokenSerializer.java).

RESThub depends on:

``` xml
  <groupId>com.fasterxml.jackson.core</groupId>
  <artifactId>jackson-*</artifactId>
```

So I think the objectmapper we're using is not picking up the `org.codehaus.jackson.map.annotate.JsonSerialize` annotation (thus not using the custom serializer); and we get this exception:

```
com.fasterxml.jackson.databind.JsonMappingException:
Can not deserialize instance of java.lang.String out of START_ARRAY token
```

because the serialized token is the actual Java object (the error refers to the "scope" attribute being an array):

``` javascript
{"value":"adf83c57-830d-4861-b057-d9a6722b4f60",
"expiration":1353448198023,
"tokenType":"bearer",
"refreshToken":null,
"scope":[],
"additionalInformation":{},
"expired":false,
"expiresIn":1}
```

**Comment by bclozel** [comment]
Opened an issue on [SpringSource JIRA - spring-security-oauth](https://jira.springsource.org/browse/SECOAUTH-358).

**Comment by bclozel** [comment]
Fixed PR #141 by commenting those tests (ugly, I know) to get those fixes into master.
